user_signup/main.py

```diff
@@ -7,18 +7,72 @@
 	loader=jinja2.FileSystemLoader(os.path.dirname(__file__)), 
 	autoescape=True)
 
-USER_RE = re.compile(r"^[a-zA-Z0-9_-]{3,20}$")
-PW_RE = re.compile(r"^.{3,20}$")
-EMAIL_RE = re.compile(r"^[\S]+@[\S]+.[\S]+$")
-
-def validate_username(username):
-	return USER_RE.match(username)
-
-def validate_password(password):
-	return PW_RE.match(password)
-
-def validate_email(email):
-	return EMAIL_RE.match(email)
+USER_RE = r"^[a-zA-Z0-9_-]{3,20}$"
+PW_RE = r"^.{3,20}$"
+EMAIL_RE = r"^[\S]+@[\S]+.[\S]+$"
+
+USER_ERROR = "Invalid username"
+PW_ERROR = "Invalid password"
+VERIFY_ERROR = "Passwords don't match"
+EMAIL_ERROR = "Invalid email address"
+
+
+def build_validation_re(USER_RE, PW_RE, EMAIL_RE):
```

Comment:
This and `build_error_messages` don't really need to be functions, you can just have it look like:
```python
VALIDATION_MATCHERS = {
  'username': re.compile(USER_RE),
  'password': re.compile(PW_RE),
  'email': re.compile(EMAIL_RE)
}
```